modules/nwtc-library/src/NWTC_C_Binding.f90

```diff
@@ -49,7 +49,73 @@ SUBROUTINE SetErr(ErrStat, ErrMsg, ErrStat_C, ErrMsg_C)
     else
         ErrMsg_C = TRANSFER( TRIM(ErrMsg)//C_NULL_CHAR, ErrMsg_C )
     endif
-END SUBROUTINE SetErr
+END SUBROUTINE SetErrStat_F2C
+
+!> This subroutine incorporates the local error status and error messages into the global error
+!! status and message. It expects both local and global error messages to be null-terminated
+!! C strings.
+SUBROUTINE SetErrStat_C(ErrStatLocal, ErrMessLocal, ErrStatGlobal, ErrMessGlobal, RoutineName)
+
+    INTEGER(C_INT),                          INTENT(IN   ) :: ErrStatLocal                  ! Error status of the operation
+    CHARACTER(KIND=C_CHAR, LEN=ErrMsgLen_C), INTENT(IN   ) :: ErrMessLocal                  ! Error message if ErrStat /= ErrID_None
+    INTEGER(C_INT),                          INTENT(INOUT) :: ErrStatGlobal                 ! Error status of the operation
+    CHARACTER(KIND=C_CHAR),                  INTENT(INOUT) :: ErrMessGlobal(ErrMsgLen_C)    ! Error message if ErrStat /= ErrID_None
+    CHARACTER(*),                            INTENT(IN   ) :: RoutineName                   ! Name of the routine error occurred in
+
+    IF ( ErrStatLocal == ErrID_None ) RETURN
+
+    IF (ErrStatGlobal /= ErrID_None) THEN
```

Comment:
leftover debugging?

Reply:
I was still working through this subroutine, but I've resolved it now.